### DISCUSSION TITLE
How to log the user out when the session has expired

### DISCUSSION THREAD

**Discussion by matine** [nextauthjs/next-auth#8200] [Help]
### Question 💬

I'm trying to understand the proper way to log out a user when the session has expired.

In our current implementation we return null in the session callback if the token has expired which works well, however we've currently switched to using typeScript strict mode and realised the callback return type doesn't actually allow null. So I'm wondering if there's a suggested method to implement this?

### How to reproduce ☕️

This is our current code which doesn't pass in strict mode:

```
import NextAuth, { NextAuthOptions } from 'next-auth';

const authOptions: NextAuthOptions = {
...
    session: ({ session, token }) => {
      session.user = token.user;
      if (new Date(token.user.customerAccessToken.expiresAt) < new Date()) {
        console.warn('Session expired');
        return null;
      }

      return session;
    },
  },
...
}
```

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by rexfordessilfie** [comment]
Hello @matine. Could you try with `undefined`? It looks like the session callback also allows undefined as a return value. I am on next-auth@4.22.1.

<img width="687" alt="Screenshot 2023-07-10 at 02 42 53" src="https://github.com/nextauthjs/next-auth/assets/43508242/e25c555c-72b6-4c6c-8d75-e4483e8f47f1">

**Comment by klee3** [comment]
Yeah, It does log user out, but don't throw the user to signIn page. 
here is my middleware code written in nextjs
```
import { withAuth } from "next-auth/middleware";

export default withAuth({
  callbacks: {
    authorized: ({ req, token }) => {
      if (token && new Date() > new Date((token as any).expires_at))
        return true;
      return false;
    },
  },
  pages: {
    signIn: "/signin",
  },
});
export const config = {
  matcher: [
    "/dashboard/:path*",
    "/server/:path*",
    "/register/:path*",
    "/api/:path*",
  ],
};
```
It do work when I use signout method on a button click but not on returning undefine
